code/obj/machinery/plantpot.dm

```diff
@@ -741,30 +717,34 @@
 			pingsignal.transmission_method = TRANSMISSION_RADIO
 
 			var/datum/radio_frequency/frequency = radio_controller.return_frequency("[report_freq]")
-			if (!frequency) return
+			if(!frequency) return
 			SPAWN_DBG(0.5 SECONDS) //Send a reply for those curious jerks
 				frequency.post_signal(src, pingsignal)
 
 		return //Just toss out the rest of the signal then I guess
 
 	// Procs specific to the plantpot start here.
 
-	proc/update_water()
+	proc/update_water_icon()
+		var/datum/color/average
 		src.water_sprite = image('icons/obj/hydroponics/hydroponics.dmi',"wat-[src.total_volume]")
 		src.water_display = image('icons/obj/hydroponics/hydroponics.dmi',"ind-wat-[src.water_level]")
 
-		if (src.reagents.total_volume)
-			var/datum/color/average = src.reagents.get_average_color()
-			src.water_sprite.color = average.to_rgba()
+		if(src.reagents.total_volume)
+			average = src.reagents.get_average_color()
+			if(average.to_rgba() != src.water_sprite.color)
+				src.water_sprite.color = average.to_rgba()
```

Comment:
Right now you're calling `to_rgba()` on `average` twice, it's probably better to just stick with the single line of `src.water_sprite.color = average.to_rgba()` without the check.

Reply:
Removed duplicate proc call in commit da26545